sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/python/GpuMapInBatchExec.scala

```diff
@@ -52,12 +52,8 @@ trait GpuMapInBatchExec extends ShimUnaryExecNode with GpuPythonExecBase {
     val (numInputRows, numInputBatches, numOutputRows, numOutputBatches,
          spillCallback) = commonGpuMetrics()
 
-    lazy val isPythonOnGpuEnabled = GpuPythonHelper.isPythonOnGpuEnabled(conf)
     val pyInputTypes = child.schema
     val chainedFunc = Seq(ChainedPythonFunctions(Seq(pandasFunction)))
-    val sessionLocalTimeZone = conf.sessionLocalTimeZone
-    val pythonRunnerConf = ArrowUtils.getPythonRunnerConfMap(conf)
-    val localOutput = output
```

Comment:
The point of these local variables was to avoid needing to serialize the entire GpuMapInBatchExec instance for the `mapPartitionsInternal` block (and thus serialize also the child nodes, effectively the rest of the plan above this node).  Accessing `conf` will require serializing the instance, so I recommend caching it in a local variable and only referencing that local variable in the serialized block below.

We also should keep `localOutput` for the same reasons.

Reply:
Local output works but the local conf seems not work as expected.
I got a NullPointerException when caching the conf.
Code:
```
val localConf = conf
rdd.mapPartitionsInternal { it =>
   ............
   val isPythonOnGpuEnabled = GpuPythonHelper.isPythonOnGpuEnabled(localConf)
   ...........  
}
```
Error
```
E                   Caused by: java.lang.NullPointerException
E                   	at org.apache.spark.sql.internal.SQLConf.getAllConfs(SQLConf.scala:4624)
E                   	at com.nvidia.spark.rapids.RapidsConf.<init>(RapidsConf.scala:1722)
E                   	at org.apache.spark.sql.rapids.execution.python.GpuPythonHelper$.isPythonOnGpuEnabled(GpuPythonHelper.scala:81)
E                   	at org.apache.spark.sql.rapids.execution.python.GpuMapInBatchExec.$anonfun$doExecuteColumnar$1(GpuMapInBatchExec.scala:71)
E                   	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:890)
E                   	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:890)
E                   	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)

```

Reply:
done